pinot-common/src/main/java/org/apache/pinot/common/utils/helix/TableCache.java

```diff
@@ -0,0 +1,189 @@
+              _tableConfigMap.put(tableNameWithType, tableConfig);
+              String rawTableName = TableNameBuilder.extractRawTableName(tableNameWithType);
+              //create case insensitive mapping
+              _tableNameMap.put(tableNameWithType.toLowerCase(), tableNameWithType);
+              _tableNameMap.put(rawTableName.toLowerCase(), rawTableName);
+              //create case insensitive mapping between table name and schemaName
+              _table2SchemaConfigMap.put(tableNameWithType.toLowerCase(), rawTableName);
+              _table2SchemaConfigMap.put(rawTableName.toLowerCase(), rawTableName);
+            } catch (IOException e) {
+              e.printStackTrace();
+              //ignore
+            }
+          }
+        }
+      } catch (Exception e) {
+        e.printStackTrace();
+        //ignore
+      }
+    }
+
+    @Override
+    public void handleChildChange(String s, List<String> list)
+        throws Exception {
+      refresh();
+    }
+
+    @Override
+    public void handleDataChange(String s, Object o)
+        throws Exception {
+      refresh();
+    }
+
+    @Override
+    public void handleDataDeleted(String s)
+        throws Exception {
+      refresh();
+    }
+  }
+
+  class SchemaChangeListener implements IZkChildListener, IZkDataListener {
+    Map<String, Schema> _schemaMap = new ConcurrentHashMap<>();
+    Map<String, Map<String, String>> _schemaColumnMap = new ConcurrentHashMap<>();
+
+    public synchronized void refresh() {
+      try {
+        //always subscribe first before reading, so that we dont miss any changes
+        _propertyStore.subscribeChildChanges(PROPERTYSTORE_SCHEMAS_PREFIX, _schemaChangeListener);
+        _propertyStore.subscribeDataChanges(PROPERTYSTORE_SCHEMAS_PREFIX, _schemaChangeListener);
+        List<ZNRecord> children =
+            _propertyStore.getChildren(PROPERTYSTORE_SCHEMAS_PREFIX, null, AccessOption.PERSISTENT);
+        if (children != null) {
+          for (ZNRecord znRecord : children) {
+            try {
+              Schema schema = SchemaUtils.fromZNRecord(znRecord);
+              String schemaNameLowerCase = schema.getSchemaName().toLowerCase();
+              _schemaMap.put(schemaNameLowerCase, schema);
+              Collection<FieldSpec> allFieldSpecs = schema.getAllFieldSpecs();
+              ConcurrentHashMap<String, String> columnNameMap = new ConcurrentHashMap<>();
+              _schemaColumnMap.put(schemaNameLowerCase, columnNameMap);
+              for (FieldSpec fieldSpec : allFieldSpecs) {
+                columnNameMap.put(fieldSpec.getName().toLowerCase(), fieldSpec.getName());
+              }
+            } catch (IOException e) {
+              //ignore
+            }
+          }
+        }
+      } catch (Exception e) {
+        e.printStackTrace();
+        //ignore
+      }
+    }
+
+    Schema getSchema(String schemaName) {
+      return _schemaMap.get(schemaName.toLowerCase());
```

Comment:
Since the other listener's map is sufficient to first get the schema name, this method and _schemaMap is probably not needed.

Reply:
yes, just had it in case we need to get schema in the future but can remove it and add it back when we need it.